localstack/utils/analytics/metadata.py

```diff
@@ -101,32 +102,58 @@ def _generate_session_id() -> str:
 
 
 def _generate_machine_id() -> str:
+    try:
+        # try to get a robust ID from the docker socket
+        from localstack.utils.docker_utils import DOCKER_CLIENT
+
+        docker_id = DOCKER_CLIENT.get_system_id()
```

Comment:
We could do something like `if DOCKER_CLIENT.get_system_id() == DOCKER_CLIENT.get_system_id():` here, to check if we are in Docker (ID reliable) or in podman (generating an new one every time, then we could as well use one of the fallbacks).

Reply:
good idea, will add